ledger/simulation/simulation_eval_test.go

```diff
@@ -647,21 +586,16 @@ int 1`,
 func TestSimpleAppCall(t *testing.T) {
 	partitiontest.PartitionTest(t)
 	t.Parallel()
+	simulationTest(t, func(accounts []simulationtesting.Account, txnInfo simulationtesting.TxnInfo) simulationTestCase {
+		sender := accounts[0]
 
-	for _, signed := range []bool{true, false} {
-		signed := signed
-		t.Run(fmt.Sprintf("signed=%t", signed), func(t *testing.T) {
-			t.Parallel()
-			simulationTest(t, func(accounts []simulationtesting.Account, txnInfo simulationtesting.TxnInfo) simulationTestCase {
-				sender := accounts[0]
-
-				// Create program and call it
-				futureAppID := basics.AppIndex(1)
-				createTxn := txnInfo.NewTxn(txntest.Txn{
-					Type:          protocol.ApplicationCallTx,
-					Sender:        sender.Addr,
-					ApplicationID: 0,
-					ApprovalProgram: `#pragma version 6
+		// Create program and call it
+		futureAppID := basics.AppIndex(1)
```

Comment:
I suppose this will break with new `tc` start. How are resource IDs chosen in simulation? I assume it's the next ID, as usual, when done on chain. I guess here the simulation is occurring on an empty chain.

Reply:
Yes, simulation should use the same mechanism for IDs as real evaluation, so I agree these will break when the initial `tc` changes